linkml_runtime/utils/schemaview.py

```diff
@@ -445,39 +448,20 @@ def ordered(self, elements: ElementDict, ordered_by: OrderedBy | None = None) ->
     def _order_lexically(self, elements: ElementDict) -> ElementDict:
         """Order elements by name.
 
+        Note: sort is case sensitive, based on the name used in the original schema.
+
         :param element: slots or class type to order
         :return: all classes or slots sorted lexically in schema view
         """
-        ordered_list_of_names = []
-        ordered_elements = {}
-        for c in elements:
-            ordered_list_of_names.append(c)
-        ordered_list_of_names.sort()
-        for name in ordered_list_of_names:
-            ordered_elements[self.get_element(name).name] = self.get_element(name)
-        return ordered_elements
+        return {name: elements[name] for name in sorted(elements.keys())}
```

Comment:
dictionaries can be relied upon to return their contents in order of insertion, so no need for all the extra gubbins in the original implementation